### PR TITLE
Update EditSketchModal.js

### PR DESCRIPTION
**Pull request by nendow02**
Refactored editSketchModal

### CODE REVIEW

**Comment by lumisphere902** [comment]
I'm still seeing a lot of lint errors... try running `git fetch origin` `git merge master` and `npm install`?

**Comment by nendow02** [comment]
Sorry I should have mentioned, I was able to get all but two lint errors dealt with. The ones I can’t do is  `error  Visible, non-interactive elements with click handlers must have at least one keyboard listener` the other is  `error  Non-interactive elements should not be assigned mouse or keyboard event listeners`. I left it without any lint fixes because of that, I’m not sure how I’m supposed to fix that error.

File: src/components/Sketches/components/EditSketchModal.js [hunk 1]
```diff
@@ -224,24 +226,22 @@ class EditSketchModal extends React.Component {
               <Button
                 color="primary"
                 size="lg"
-                onClick={() => {
-                  this.setState({ onThumbnails: true });
-                }}
+                onClick={() => setOnThumbnails(true)}
```
Comment (lumisphere902) [inline src/components/Sketches/components/EditSketchModal.js:229]: For this and the other `onClick` handlers, you should add an `onKeyDown` and `tabIndex` attribute

File: src/components/Sketches/components/EditSketchModal.js [hunk 2]
```diff
@@ -264,30 +264,26 @@ class EditSketchModal extends React.Component {
...
-          />
-        </figure>
-      );
-    });
+  const renderThumbnailModal = () => {
+    const icons = ThumbnailArray.map((val, index) => (
+      <figure
+        className="sketches-gallery-item"
+        key={val}
+        onClick={() => setNewThumbnail(index)}
```
Comment (lumisphere902) [inline src/components/Sketches/components/EditSketchModal.js:272]: `figure` is not an interactive element by default, so you could add `role="button"` to mark it as interactive

**Review by lumisphere902 — CHANGES_REQUESTED** [review]
Hopefully the below comments help with your lint issues

File: src/components/Sketches/components/EditSketchModal.js (hunk 2 above)
Comment (nendow02) [reply src/components/Sketches/components/EditSketchModal.js:272]: Used `role="presentation"` instead, for some reason "button" didn’t work

File: src/components/Sketches/components/EditSketchModal.js [hunk 3]
```diff
@@ -1,181 +1,180 @@
...
+    setNewName(-1);
+    setNewThumbnail(-1);
+    setError('');
+    setOnThumbnails(false);
+    setDisableSubmit(false);
   };
 
-  changeLanguage = (lang) => {
-    this.setState({ newLanguage: lang });
+  const changeLanguage = (lang) => {
```
Comment (lumisphere902) [inline src/components/Sketches/components/EditSketchModal.js:48]: You can get rid of this and just use setNewLanguage

File: src/components/Sketches/components/EditSketchModal.js [hunk 4]
```diff
@@ -1,181 +1,180 @@
...
+              setDisableSubmit(false);
+              setError(res.text() || 'Failed to edit sketch, please try again later');
             }
           })
-          .catch((err) => {
-            this.setState({
-              disableSubmit: false,
-              error: 'Failed to edit sketch, please try again later',
-            });
-            console.log(err);
```
Comment (lumisphere902) [inline src/components/Sketches/components/EditSketchModal.js:144]: Replace these with `console.error`

File: src/components/Sketches/components/EditSketchModal.js [hunk 5]
```diff
@@ -224,24 +219,26 @@ class EditSketchModal extends React.Component {
               <Button
                 color="primary"
                 size="lg"
-                onClick={() => {
-                  this.setState({ onThumbnails: true });
-                }}
+                onClick={() => setOnThumbnails(true)}
+                onKeyDown={() => {}}
```
Comment (lumisphere902) [inline src/components/Sketches/components/EditSketchModal.js:223]: The keydown function should be equivalent to the click function; the purpose is to aid those who can’t use a mouse

**Review by lumisphere902 — CHANGES_REQUESTED** [review]
Good changes; just a few more small fixes and it should be good

**Review by reginawang3495 — APPROVED** [review]
looks good to me!! thanks for making the changes :))

**Review by lumisphere902 — APPROVED** [review]
LGTM. Thanks for all the work!

**Pull request merged**